src/Compiler/TypedTree/tainted.fs

```diff
@@ -101,11 +101,12 @@ type internal Tainted<'T> (context: TaintedContext, value: 'T) =
             | :? TypeProviderError -> reraise()
             | :? AggregateException as ae ->
                     let errNum,_ = FSComp.SR.etProviderError("", "")
-                    let messages = [for e in ae.InnerExceptions -> e.Message]
+                    let messages = [for e in ae.InnerExceptions -> if e.InnerException = null then e.Message else (e.Message + ": " + e.GetBaseException().Message)]
                     raise <| TypeProviderError(errNum, this.TypeProviderDesignation, range, messages)
             | e -> 
                     let errNum,_ = FSComp.SR.etProviderError("", "")
-                    raise <| TypeProviderError((errNum, e.Message), this.TypeProviderDesignation, range)
+                    let error = if e.InnerException = null then e.Message else (e.Message + ": " + e.GetBaseException().Message)
+                    raise <| TypeProviderError((errNum, error), this.TypeProviderDesignation, range)
```

Comment:
Same here (it's never null):
```suggestion
                    raise <| TypeProviderError((errNum, e.GetBaseException().Message), this.TypeProviderDesignation, range)
```

Reply:
Changed to isNull. I remember seeing some weird cases in the past where "GetBaseException()" has failed for null.

Reply:
@Thorium, sure it fails for `null`, but you're already catching an exception, so `e` is never null. This works, for instance: `(exn "foo bar").GetBaseException()`. Also, check the docs and implementation, which say the same thing.